sleuth/sleuth/enrich_tls.py

```diff
@@ -89,9 +102,102 @@ def check_compliance(compliance_policy, scs):
                 return "yes" if scs_desc in compliance_data[compliance_policy] else "no"
 
     return "error loading file"
+    
+    
+def audit_certs_issuer(certs, trusted_ca_list):
+    # as implemented now, we will get the tls certs in order so that the root
+    # cert is at the end. We check to see if the issuer is in our trusted_ca_list;
+    # if not, we report it as a concern
+    for each in certs[-1]['issuer']:
+        if each['entry_id'] == 'organizationName':
+            if each['entry_data'] not in trusted_ca_list:
+                return 'CA not trusted: ' + each['entry_data']
+    
+    return None
+
+
+def get_seclevel_with_param(policy, attr, value, param):
+    current_seclevel = None
+
+    if param:
```

Comment:
Shouldn't need this check because param is not optional to the function.

Reply:
It's not optional, but there are cases when it can be passed in as 'None'. For example, if "kex" is "RSA" but we don't have the key size info. I wanted the logic for this to stay in this function because regardless of whether or not the param exists, we are evaluating an attribute that has additional conditions. But, I will move this around a bit so it makes more sense!